Automattic/wp-calypso
# Migrate the requestSiteAlerts data getter to React Query

Migrates the `requestSiteAlerts` data getter to React Query. It's used in Activity Log to fetch info about site's theme and core updates:

<img width="621" alt="Screenshot 2022-01-24 at 10 18 15" src="https://user-images.githubusercontent.com/664258/150768769-2081e898-47b5-464c-9d1f-d1a717fc22d7.png">

There is one tricky place when migrating the `WithItemsToUpdate` to a functional component: it has a `getDerivedStateFromProps` method that merges the versions of `pluginsWithUpdates` into the current object (list of plugins keyed by slug). So that when a plugin used to have an update in one selector call no longer has an update in a later selector call (the plugin has been updated in the meantime), then the plugin is not removed from the list, but stays there.

I rewrote this using a ref that does the merge on each render. The reason why I'm using ref instead of state is that I don't need a state update when modifying the merged list: the state update was already triggered by Redux change.

**How to test:**
Find a site with pending updates (tricky task now when autoupdates are almost everywhere) and verify the list of updates is rendered in Activity Log. Try to update a plugin or a theme or Core and verify that the item is still rendered in the list until the update finishes.
